kernel/src/main/java/org/kframework/utils/options/OuterParsingOptions.java

```diff
@@ -24,20 +24,20 @@ public OuterParsingOptions() {}
     public OuterParsingOptions(Void v) {}
 
     public OuterParsingOptions(File mainDefinitionFile) {
-      this.mainDefinitionFile = mainDefinitionFile;
+        this.mainDefinitionFile = mainDefinitionFile;
     }
 
     @Parameter(description="<file>")
-    private List<String> parameters;
+    private String mainParameter;
 
     private File mainDefinitionFile;
 
     public synchronized File mainDefinitionFile(FileUtil files) {
         if (mainDefinitionFile == null) {
-            if (parameters == null || parameters.size() == 0) {
+            if (mainParameter == null) {
```

Comment:
Does this `if` ever trigger?
Judging by your test, it doesn't.

Reply:
It does trigger; the test invocation is:
```console
$ make -C bad-flags 
make: Entering directory '/home/bruce/code/k/k-distribution/tests/regression-new/bad-flags'
/home/bruce/code/k/k-distribution/bin/kompile 2>&1 | diff - no-flags.out 
/home/bruce/code/k/k-distribution/bin/kompile --syntax-module TEST --no-exc-wrap --badflag --extra --backend llvm  test.k -d . 2>&1 | diff - extra-flags.out 
make: Leaving directory '/home/bruce/code/k/k-distribution/tests/regression-new/bad-flags'
```

The contents of the "no flags supplied" test is:
```console
$ cat bad-flags/no-flags.out 
[Error] Critical: You have to provide exactly one main file in order to do
outer parsing.
```

Reply:
You are right, I wasn't paying attention.